apps/api/migrations/integration-scheme-update/add-primary-priority-migration.ts

```diff
@@ -0,0 +1,139 @@
+    if (!organization) {
+      console.log(
+        `Organization ${environment._organizationId} is not found for environment ${environment.name}, id: ${environment._id}`
+      );
+      continue;
+    }
+
+    console.log('\n------------------------------------------');
+    console.log(
+      `Migrating integrations for the "${organization.name}" organization in the ${environment.name} environment:`
+    );
+
+    for (const channel of CHANNELS_WITH_PRIMARY) {
+      const integrations = await integrationRepository.find(
+        {
+          _organizationId: organization._id,
+          _environmentId: environment._id,
+          channel,
+          active: true,
+        },
+        undefined,
+        { sort: { createdAt: -1 } }
+      );
+      console.log('------');
+      console.log(`Found ${integrations.length} active integrations for the ${channel} channel`);
+
+      for (let i = 0; i < integrations.length; i++) {
+        const integration = integrations[i];
+        const payload = {
+          primary: i === 0,
+          priority: integrations.length - i,
+        };
+        console.log(`${i + 1}. Updating integration "${integration.name}" with: `, payload);
+        await integrationRepository.update(
+          {
+            _id: integration._id,
+            _environmentId: integration._environmentId,
+            _organizationId: integration._organizationId,
+          },
+          {
+            $set: payload,
+          }
+        );
+      }
+
+      const inactiveIntegrations = await integrationRepository.find({
+        _organizationId: organization._id,
+        _environmentId: environment._id,
+        channel,
+        active: false,
+      });
+      console.log('------');
+      console.log(`Found ${inactiveIntegrations.length} inactive integrations for the ${channel} channel`);
+
+      for (let i = 0; i < inactiveIntegrations.length; i++) {
+        const integration = inactiveIntegrations[i];
+        const payload = {
+          primary: false,
+          priority: 0,
+        };
+        console.log(`${i + 1}. Updating inactive integration "${integration.name}" with: `, payload);
+        await integrationRepository.update(
+          {
+            _id: integration._id,
+            _environmentId: integration._environmentId,
+            _organizationId: integration._organizationId,
+          },
+          {
+            $set: payload,
+          }
+        );
+      }
+    }
+
+    for (const channel of [ChannelTypeEnum.IN_APP, ChannelTypeEnum.PUSH, ChannelTypeEnum.CHAT]) {
+      const integrations = await integrationRepository.find({
+        _organizationId: organization._id,
+        _environmentId: environment._id,
+        channel,
+      });
+      console.log('------');
+      console.log(`Found ${integrations.length} integrations for the ${channel} channel`);
+
+      for (let i = 0; i < integrations.length; i++) {
+        const integration = integrations[i];
```

Comment:
what do you think about doing a small check if primary/priority already exists in the integration and if not skip the update?

Reply:
@LetItRock

Reply:
@djabarovgeorge for the above channels there won't be primary integrations... but if you mean for the sms/email and you are worried that this migration might fail, I would say that we will run it in the beginning without the updates just to get some statistics and see if it will pass, and if everything will be ok then we will run it with updates... 

anyway, even if it will fail it's "safe" to run it a second time, because the integrations are updated by order of `createdAt` desc...

Reply:
I am not afraid of failure just for side effects for the user if it will time from the fist and second run. But if the difference will be small there shouldn't be issue.

Reply:
we will figure it out :)